File: evidence/generated-systems/golang/governance-on-golang/governance-on-golang/internal/model/DataBreach.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// DataBreach Declaration
//==============================================================
type DataBreach struct {
    gorm.Model
     IncidentDate                                                            time.Time
    Description                                    string
    RecordsAffected                                                            string
    NotificationRequired                                    bool
    OrganizationId         *uint
    Organization           *Organization `gorm:"foreignKey:OrganizationId"`
     ProcessingActivities           []DataProcessingActivity `gorm:"foreignKey:ProcessingActivitiesFromDataBreachId"`
     DataCategories           []DataCategory `gorm:"foreignKey:DataCategoriesFromDataBreachId"`
     ThirdParties           []ThirdParty `gorm:"foreignKey:ThirdPartiesFromDataBreachId"`
    MatterId         *uint
    Matter           *Matter `gorm:"foreignKey:MatterId"`
    Severity                      BreachSeverity
    Status                      IncidentStatus

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/governance-on-rails/governanceonrails/app/models/databreach.rb
class DataBreach < ApplicationRecord
  enum Severity: [:Low, :Medium, :High, :Critical]
  enum Status: [:Identified, :Contained, :Notified, :Resolved, :Closed]


  has_many :Organization, class_name: 'Organization'
  has_many :ProcessingActivities, class_name: 'DataProcessingActivity'
  has_many :DataCategories, class_name: 'DataCategory'
  has_many :ThirdParties, class_name: 'ThirdParty'
  has_many :Matter, class_name: 'Matter'

end
